dotnet/src/Connectors/Connectors.UnitTests/OpenAI/TextToImage/AzureOpenAITextToImageTests.cs

```diff
@@ -13,17 +14,60 @@
 namespace SemanticKernel.Connectors.UnitTests.OpenAI.TextToImage;
 
 /// <summary>
-/// Unit tests for <see cref="AzureOpenAITextToImageTests"/> class.
+/// Unit tests for <see cref="AzureOpenAITextToImageServiceTests"/> class.
 /// </summary>
-public sealed class AzureOpenAITextToImageTests
+public sealed class AzureOpenAITextToImageServiceTests
 {
+    [Theory]
+    [InlineData("My deployment", "dall-e-3", true, null)]
+    [InlineData("My deployment", "dall-e-2", false, typeof(NotSupportedException))]
+    [InlineData("My deployment", "Dall-E-2", false, typeof(NotSupportedException))]
+    [InlineData("", "", false, typeof(ArgumentException))]
+    public void ItValidatesTheModelId(string deploymentName, string modelId, bool shouldBeValid, Type expectedExceptionType)
+    {
+        // Act
+        if (!shouldBeValid)
+        {
+            Assert.Throws(expectedExceptionType, () => new AzureOpenAITextToImageService(deploymentName: deploymentName, modelId: modelId, endpoint: "https://az.com", apiKey: "abc"));
+        }
+        else
+        {
+            // No exceptions for these
+            new AzureOpenAITextToImageService(deploymentName: deploymentName, modelId: modelId, endpoint: "https://az.com", apiKey: "abc");
+        }
+    }
+
+    [Fact(Skip = "Needs refactoring, decouple from Azure SDK implementation details")]
+    public async Task ItGeneratesImagesAsync()
```

Comment:
Should we remove the test in this case?

Reply:
Removed this and added other for the NotSupportedException scenarios.